frontend/model/captureLogs.js

```diff
@@ -0,0 +1,176 @@
+  localStorage.setItem('giConsole/count', entriesCount)
+
+  verifyLogsSize()
+
+  // To avoid infinite loop because we log all selector calls, we run sbp calls
+  // here in a roundabout way by getting the function to which they're mapped.
+  // The reason this works is because the entire `sbp` domain is blacklisted
+  // from being logged in main.js.
+  sbp('sbp/selectors/fn', 'okTurtles.events/emit')(CAPTURED_LOGS, lastEntry)
+}
+
+function verifyLogsSize () {
+  if (entriesCount % ENTRIES_MARKER === 0) {
+    const markers = getMarkers()
+    // Save entry as a marker to be later deleted when logs are too big.
+    markers.push(lastEntry)
+    localStorage.setItem('giConsole/markers', JSON.stringify(markers))
+  }
+
+  if (entriesCount >= ENTRIES_LIMIT) {
+    const markers = getMarkers()
+    let toDelete = ENTRIES_MARKER // delete the latest x entries
+    let prevEntry = markers.shift()
+    do {
+      const entry = JSON.parse(localStorage.getItem(`giConsole/${prevEntry}`)) || {}
+      localStorage.removeItem(`giConsole/${prevEntry}`)
+      prevEntry = entry.prev
+      entriesCount--
+      toDelete--
+    } while (toDelete && prevEntry)
+    localStorage.setItem('giConsole/markers', JSON.stringify(markers))
+    localStorage.setItem('giConsole/count', entriesCount)
+  }
+}
+
+export function captureLogsStart () {
+  console.debug('captureLogsStart called')
+  isCapturing = true
+  lastEntry = localStorage.getItem('giConsole/lastEntry')
+  entriesCount = +localStorage.getItem('giConsole/count') || 0
+
+  // Set a new visit or session - useful to understand logs through time.
+  // NEW_SESSION -> The user opened a new browser or tab.
+  // NEW_VISIT -> The user comes from an ongoing session (refresh or login)
+  const isNewSession = !sessionStorage.getItem('NEW_SESSION')
+  if (isNewSession) { sessionStorage.setItem('NEW_SESSION', 1) }
+  captureLog(isNewSession ? 'NEW_SESSION' : 'NEW_VISIT')
+
+  // Subscribe to appLogsFilter
+  const state = sbp('state/vuex/state')
+  appLogsFilter = state.appLogsFilter || []
+  sbp('okTurtles.events/on', SET_APP_LOGS_FILTER, filter => { appLogsFilter = filter })
+
+  console.debug('Starting to capture logs of type:', appLogsFilter)
+
+  // Override the console to start capturing the logs
+  if (!isConsoleOveridden) {
+    // avoid duplicated captures, in case captureLogsStart
+    // is called multiple times. (ex: login twice in the same visit)
+    isConsoleOveridden = true
+
+    // NOTE: Find a way to capture logs without messing up with log file location.
+    // console.log() doesnt include stack trace, so when logged, we can't access
+    // where the log came from (file name), which difficults debugging if needed.
+    window.console = new Proxy(window.console, {
+      get: (obj, type) => (...args) => {
+        obj[type](...args)
+        if (isCapturing && appLogsFilter.includes(type)) {
+          captureLog(type, ...args)
+        }
```

Comment:
Let's sync the checkboxes with real console logs as well. In `process.env.NODE_ENV === 'development'` enable all checkboxes by default in `state.js` default vuex settings.

So here let's make the condition look like this:

```js
        if (appLogsFilter.includes(type)) {
          obj[type](...args)
          isCapturing && captureLog(type, ...args)
        }
```

Reply:
Doing the condition that you are suggesting will cause the unfiltered logs to not be fired at all. `obj[type](...args)` (the original log) must always be fired, even if it's not part of the filters and even if we enable all the checkboxes by default. If then we change a filter (ex: uncheck `warn`), all consequent `console.warn` won't be fired or captured. We still want them to be fired, that's the purpose of `obj[type](...args)` and that's why it should be outside the condition.

Anyway... I changed the filters to be all checked by default during development but kept the condition as it is.

Just for curiosity, why do you wanna check all filters during dev?

Reply:
> obj[type](...args) (the original log) must always be fired,

No, we do not want it fired in production for anything except warn and error. And for warn and error I already addressed that in another comment by saying the checkboxes in the UI must be checked and disabled (so that you cannot not log warn and error).

> Just for curiosity, why do you wanna check all filters during dev?

That's how it is right now before this PR is merged. That's how it should remain afterward. We want all console info available during dev.

Reply:
> No, we do not want it fired in production.

But that's Grunt's job (with [Uglify](https://github.com/mishoo/UglifyJS2)) to stripe out the consoles from the bundle. It shouldn't be done here.

> We want all console info available during dev.

And the current code is doing that regardless of filters. One thing is capturing logs, another is firing them... I think you are mixing up stuff. If you want, we can better clarify this on a call.

Reply:
> But that's Grunt's job (with Uglify) to stripe out the consoles from the bundle. It shouldn't be done here.

If Grunt strips out all of the logging, then if something goes wrong we won't be able to ask users to re-enable logging and send it to us.

> And the current code is doing that regardless of filters. One thing is capturing logs, another is firing them... I think you are mixing up stuff. If you want, we can better clarify this on a call.

Sure, we can do a call!

Reply:
Ups, something happened and my comment wasn't sent. I said (before your last comment) that in the meantime I finally understood what you were trying to say.
I thought the goal of this task was to only capture the logs (error, warning and debug). But then I realized you also want to "control" when the console is fired in production and in development.

Code changed.